src/NServiceBus.Core/Transports/Msmq/MsmqMessageSender.cs

```diff
@@ -34,6 +34,12 @@ public void Send(TransportMessage message, SendOptions sendOptions)
         {
             var address = sendOptions.Destination;
             var queuePath = NServiceBus.MsmqUtilities.GetFullPath(address);
+
+            if (message.TimeToBeReceived < MessageQueue.InfiniteTimeout && WillUseTransactionThatSupportsMultipleOperations(sendOptions))
```

Comment:
Can we somehow extract the result of `GetTransactionTypeForSend` so that it is not invoked twice. I know this is not a performance problematic method but my OCD...